src/Microsoft.OData.Client/ALinq/ResourceBinder.cs

```diff
@@ -2910,6 +2988,46 @@ internal static void ValidateExpandPath(Expression input, DataServiceContext con
 
                 throw new NotSupportedException(Strings.ALinq_InvalidExpressionInNavigationPath(input));
             }
+
+            /// <summary>
+            /// Checks whether the specified <paramref name="expr"/> is a valid aggregate expression.

```

Comment:
It would be good to know what counts as a valid aggregate expression.

Reply:
In the specification, the aggregate expression has to meet [one of 5 criteria](http://docs.oasis-open.org/odata/odata-data-aggregation-ext/v4.0/cs02/odata-data-aggregation-ext-v4.0-cs02.html#_Toc435016568). I'll see how I can improve the docstring to give the user a better idea. The exception message tries to capture what is wrong with the exception.

Reply:
Have to admit I can't say I fully understand this line :)
>An aggregate expression must evaluate to a single-valued property path to an aggregatable property

Reply:
@odero 
Consider the following class:
```c#
class Sale
{
    public string Currency { get; set; }
    public Amount Amount { get; set; }
    public Product Product { get; set; }
    public Collection<Review> Reviews { get; set; }
}
```
- `Currency`, `Amount` and `Product` are single-valued properties. `Reviews` is a collection property.
- On the subject of aggregatable (or is it aggregable?), only `Amount` fits the bill - think `average`, `sum`, `min`, and `max`. You can't aggregate `Currency` - of type string, or `Product` - of type Product.
- However, `Currency` would qualify as aggregatable property when using `countdistinct` aggregation.

The fact that I have used this many words and an illustration to explain tells you how hard it is to come up with a good docstring for the method :)
However, if you go through the method there are helpful comments that cover the scenarios we're validating for.